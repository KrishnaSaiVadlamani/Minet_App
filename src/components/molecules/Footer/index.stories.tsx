import { ComponentMeta, ComponentStory } from '@storybook/react'
import React from 'react'
import Footer from './index'

export default {
  title: 'Molecules/Footer',
  component: Footer,
} as ComponentMeta<typeof Footer>

const Template: ComponentStory<typeof Footer> = () => (
  <Footer/>
)

export const SimpleFooter = Template.bind({})
