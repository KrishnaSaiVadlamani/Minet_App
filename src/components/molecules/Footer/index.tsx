

import { Grid, styled, Stack, Box } from '@mui/material'
import React from 'react'
import theme from '../../../theme/index'
import ButtonComponent from '../../atoms/Button'
import TypographyComponent from '../../atoms/Typography'
import DropDownComponent from '../Dropdown'

const StyledBox = styled(Box)({
  [theme.breakpoints.up('lg')]:{
    maxWidth: '1500px'
  },
  [theme.breakpoints.up('xl')]: {
    maxWidth: '98vw'
  },
  [theme.breakpoints.down('lg')]: {
    minWidth: '1366px'
  },
  [theme.breakpoints.between('lg', 'xl')]: {
    width: '97vw',
  },
 
  paddingLeft: '104px',
})

const MainContainer = styled(Grid)({
  height: '90px',
  borderTop: `1px solid ${theme.palette.greyColors.grey100}`,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  paddingRight: '22px'
})

const StyledTypographyBlue = styled(TypographyComponent)({
  color: theme.palette.primary.primary500,
})

const StyledTypographyBlack = styled(TypographyComponent)({
  color: theme.palette.textColor.highEmphasis,
})

const StyledButton = styled(ButtonComponent)({
  width: '109px',
  height: '42px',
  border: `1px solid ${theme.palette.primary.main}`,
  fontWeight: 'bold',
  '&:hover': {
    backgroundColor: theme.palette.structural.main,
  },
})

const RightContainer = styled(Grid)({
  alignItems: 'flex-end',
  margin: '0px',
})

const Footer = () => {
  return (
    <StyledBox data-testid='footer'>
      <MainContainer container>
        <Grid item xs={6}>
          <Stack direction="row" spacing={3}>
            <StyledTypographyBlue variant="body2" children={'Dashboard'} />
            <StyledTypographyBlue variant="body2" children={'Careers'} />
            <StyledTypographyBlue
              variant="body2"
              children={'Legal & Privacy'}
            />
            <StyledTypographyBlack variant="body2" children={'Â© 2021 Minet'} />
          </Stack>
        </Grid>
        <RightContainer item xs={6} justifyContent="flex-end">
          <Stack direction="row" spacing={2} justifyContent="flex-end">
            <DropDownComponent onChange={() => {}} menuList={['English']} width={'170px'}/>
            <StyledButton variant="outlined" children="NEED HELP" backgroundColor={'none'}/>
          </Stack>
        </RightContainer>
      </MainContainer>
    </StyledBox>
  )
}

export default Footer
