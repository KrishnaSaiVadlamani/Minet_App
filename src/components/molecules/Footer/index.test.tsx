import renderer from 'react-test-renderer';
import Footer from '.';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';

it('renders footer correctly', () => {
    const tree = renderer
      .create(<Footer/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
});

describe('Molecules/Footer', () => {
    test('Renders footer correctly', () => {
        render(<Footer />);
        //screen.getByRole('')
        expect(screen.getByTestId('footer')).toBeInTheDocument();
    })
})