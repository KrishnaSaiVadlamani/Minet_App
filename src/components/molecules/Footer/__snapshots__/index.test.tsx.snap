// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders footer correctly 1`] = `
<div
  className="MuiBox-root css-vhpt6z"
  data-testid="footer"
>
  <div
    className="MuiGrid-root MuiGrid-container css-1xut4gs-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        className="css-u10pvi-MuiStack-root"
      >
        <p
          className="MuiTypography-root MuiTypography-body2 css-9syhwr-MuiTypography-root"
          data-testid="Hello"
        >
          Dashboard
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-9syhwr-MuiTypography-root"
          data-testid="Hello"
        >
          Careers
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-9syhwr-MuiTypography-root"
          data-testid="Hello"
        >
          Legal & Privacy
        </p>
        <p
          className="MuiTypography-root MuiTypography-body2 css-1iykimn-MuiTypography-root"
          data-testid="Hello"
        >
          © 2021 Minet
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-k97rmi-MuiGrid-root"
    >
      <div
        className="css-1lnp4wh-MuiStack-root"
      >
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary css-1oufaa8-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          data-testid="dropDown"
          onClick={[Function]}
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <p
              className="MuiTypography-root MuiTypography-body1 css-4umqm8-MuiTypography-root"
              data-testid="Hello"
            >
              English
            </p>
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            tabIndex={-1}
            value="English"
          />
          <div
            className="MuiBox-root css-17t8bxi"
          >
            <div>
              <img
                alt="icon"
                className="css-12z3ws7"
                data-testid="iconComponent"
                height="13px"
                src={
                  {
                    "process": [Function],
                  }
                }
                width="13px"
              />
            </div>
          </div>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-ihdtdm"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-b63ts3-MuiButtonBase-root-MuiButton-root"
          data-testid="button"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          NEED HELP
        </button>
      </div>
    </div>
  </div>
</div>
`;
